packages/hardhat-chai-matchers/src/changeEtherBalances.ts

```diff
@@ -0,0 +1,125 @@
+          ),
+          () => {
+            const lines: string[] = [];
+            actualChanges.forEach((change, i) => {
+              if (!change.eq(BigNumber.from(balanceChanges[i]))) {
+                lines.push(
+                  `Expected ${accountAddresses[i]} (address #${i} in the list) to change balance by ${balanceChanges[i]} wei, but it has changed by ${change} wei`
+                );
+              }
+            });
+            return lines.join("\n");
+          },
+          () => {
+            const lines: string[] = [];
+            actualChanges.forEach((change, i) => {
+              if (change.eq(BigNumber.from(balanceChanges[i]))) {
+                lines.push(
+                  `Expected ${accountAddresses[i]} (address #${i} in the list) not to change balance by ${balanceChanges[i]} wei`
+                );
+              }
+            });
+            return lines.join("\n");
+          },
+          balanceChanges.map(
+            (balanceChange, i) =>
+              `${
+                accountAddresses[i]
+              } balance changed by ${balanceChange.toString()} wei`
+          ),
+          actualChanges.map(
+            (actualChange, i) =>
+              `${
+                accountAddresses[i]
+              } balance changed by ${actualChange.toString()} wei`
+          )
+        );
+      });
+      this.then = derivedPromise.then.bind(derivedPromise);
+      this.catch = derivedPromise.catch.bind(derivedPromise);
+      this.promise = derivedPromise;
+      return this;
+    }
+  );
+}
+
+export async function getBalanceChanges(
+  transaction:
+    | providers.TransactionResponse
+    | Promise<providers.TransactionResponse>
+    | (() =>
+        | Promise<providers.TransactionResponse>
+        | providers.TransactionResponse),
+  accounts: Array<Account | string>,
+  options: BalanceChangeOptions
+) {
+  let txResponse: providers.TransactionResponse;
+
+  if (typeof transaction === "function") {
+    txResponse = await transaction();
+  } else {
+    txResponse = await transaction;
+  }
+
+  const txReceipt = await txResponse.wait();
+  const txBlockNumber = txReceipt.blockNumber;
+
+  const balancesAfter = await getBalances(accounts, txBlockNumber);
+  const balancesBefore = await getBalances(accounts, txBlockNumber - 1);
+
+  const txFees = await getTxFees(accounts, txResponse, options);
+
+  return balancesAfter.map((balance, ind) =>
+    balance.add(txFees[ind]).sub(balancesBefore[ind])
+  );
+}
+
+async function getTxFees(
+  accounts: Array<Account | string>,
+  txResponse: providers.TransactionResponse,
+  options: BalanceChangeOptions
+) {
+  return Promise.all(
+    accounts.map(async (account) => {
+      if (
+        options?.includeFee !== true &&
+        (await getAddressOf(account)) === txResponse.from
+      ) {
+        const txReceipt = await txResponse.wait();
+        const gasPrice = txResponse.gasPrice ?? txReceipt.effectiveGasPrice;
```

Comment:
See the previous comment for an explanation.

```suggestion
        const gasPrice = txReceipt.effectiveGasPrice ?? txResponse.gasPrice;
```